test/james.data.constraints.tests.pas

```diff
@@ -109,11 +109,11 @@ procedure TDataStreamTest.AsString;
   Buf := TMemoryStream.Create;
   Ss := TStringList.Create;
   try
-    Buf.WriteBuffer(TXT[1], Length(TXT) * SizeOf(Char));
+    Buf.WriteBuffer(AnsiString(TXT)[1], Length(TXT));
```

Comment:
@nunopicado I don't know why you needed to do this. 
Could you test something for us in Delphi but using the previously code? Instead of using `const`, declare a variable `s: string`, put the value into and test again. Maybe both compilers could do some casting to work with Unicode properly.